From df3242473bb5c1b45a9c351e33ceb1c34e97718e Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 12 Jun 2014 09:19:28 +0100
Subject: [PATCH 225/351] mfd: arizona: Use memset to clear out_mono array

Some compilers seem to generate a warning for the newer style C
construct, switch to a memset just to be safe.

Change-Id: I4926b327e6b89c27bc7a3fcc70fe243aa3e173ad
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/mfd/arizona-core.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/arizona-core.c b/drivers/mfd/arizona-core.c
index e049687..493c932 100644
--- a/drivers/mfd/arizona-core.c
+++ b/drivers/mfd/arizona-core.c
@@ -841,9 +841,11 @@ static int arizona_of_get_micbias(struct arizona *arizona,
 static int arizona_of_get_core_pdata(struct arizona *arizona)
 {
 	struct arizona_pdata *pdata = &arizona->pdata;
-	u32 out_mono[ARRAY_SIZE(pdata->out_mono)] = {0};
+	u32 out_mono[ARRAY_SIZE(pdata->out_mono)];
 	int i;
 
+	memset(&out_mono, 0, sizeof(out_mono));
+
 	pdata->reset = arizona_of_get_named_gpio(arizona, "wlf,reset", true);
 
 	arizona_of_get_micd_ranges(arizona, "wlf,micd-ranges");
-- 
1.7.9.5

